src/App.tsx

```diff
@@ -1,24 +1,253 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useEffect, useMemo, useRef, useState,
+} from 'react';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
+import { Filter } from './enums/filter';
+import { ErrorMessage } from './enums/error';
+import { Error } from './components/Error';
+import { TodoList } from './components/TodoList';
+import {
+  getTodos,
+  addTodos,
+  deleteTodo,
+  updateTodos,
+} from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 10567;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterValue, setFilterValue] = useState(Filter.All);
+  const [errorMessage, setErrorMessage]
+  = useState<ErrorMessage>(ErrorMessage.NONE);
+  const [value, setValue] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletedTodosId, setDeletedTodosId] = useState<number[] | []>([]);
+  const [activeFormInput, setActiveFormInput] = useState(false);
+  const formInputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (formInputRef.current && activeFormInput) {
+      formInputRef.current.focus();
+    }
+
+    setActiveFormInput(false);
+  }, [todos]);
+
+  const fetchData = async () => {
+    try {
+      const response = await getTodos(USER_ID);
+      const todosData = response as Todo[];
+
+      setTodos(todosData);
+    } catch (error) {
+      setErrorMessage(ErrorMessage.LOAD);
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  useEffect(() => {
+    setTimeout(() => (
+      setErrorMessage(ErrorMessage.NONE)
+    ), 3000);
+  }, [errorMessage]);
+
+  const handleDeleteTodo = (id: number) => {
+    setDeletedTodosId([id]);
+
+    deleteTodo(id)
+      .then(() => {
```

Comment:
I noticed that you use 2 different approaches in the project `async/await` and `then` for working with async functions. It's  generally recommended to be consistent and use a single approach throughout a project